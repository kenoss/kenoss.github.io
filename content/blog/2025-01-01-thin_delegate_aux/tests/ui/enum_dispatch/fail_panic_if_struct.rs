#[enum_dispatch::enum_dispatch]
trait ShapeI {
    fn area(&self) -> usize;
}

struct Rect {
    width: usize,
    height: usize,
}

#[enum_dispatch::enum_dispatch(ShapeI)]
struct Shape(Rect);

impl ShapeI for Rect {
    fn area(&self) -> usize {
        self.width * self.height
    }
}

fn main() {}
