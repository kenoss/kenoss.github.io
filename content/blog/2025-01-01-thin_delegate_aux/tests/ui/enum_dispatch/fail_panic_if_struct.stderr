error: custom attribute panicked
  --> tests/ui/enum_dispatch/fail_panic_if_struct.rs:11:1
   |
11 | #[enum_dispatch::enum_dispatch(ShapeI)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: ()
